src/main/jbake/content/blog/2017/06/9/opentracing-spring-boot.adoc

```diff
@@ -0,0 +1,147 @@
+
+In this demo series we are going to look at how simple it is to instrument various Java frameworks using
+http://opentracing.io[OpenTracing]. You will see that it requires minimal changes to the
+application code. In the last demo we will have microservice apps deployed on Kubernetes and
+all services will be traced with OpenTracing complaint tracing system.
+
+In this first demo we are going to develop and trace simple Spring Boot app.
+
+== Create a web application
+First let's write a simple web app, no let's generate it! All we have to do is just to select a web dependency.
+
+ifndef::env-github[]
+image::/img/blog/2017/spring-initializr.png[caption="Figure 1: ", title="Spring boot generator."]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/spring-initializr.png[caption="Figure 1: ", title="Spring boot generator."]
+endif::[]
+
+Now the application is generated, but it does not contain any web controller. We are going to implement a simple
+controller with two methods. One will return hello world string and the other creates HTTP request which calls
+hello endpoint. This demonstrates simple request chaining between services. Do not worry all the code is
+on https://github.com/pavolloffay/opentracing-java-examples[GitHub]. At the bottom of this article you will
+find all necessary links.
+
+.Hello Controller:
+----
+@RestController
+public class HelloController {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @RequestMapping("/hello")
+    public String hello() {
+        return "Hello from Spring Boot!";
+    }
+
+    @RequestMapping("/chaining")
+    public String chaining() {
+        ResponseEntity<String> response = restTemplate.getForEntity("http://localhost:8080/hello", String.class);
+        return "Chaining + " + response.getBody();
+    }
+}
+----
+
+Now the application can serve requests for URLs `http://localhost:8080/hello` and `http://localhost:8080/chaining`.
+The app is still not instrumented, we won't see any data coming to a tracing system.
+
+== Instrumentation
+Instrumentation with OpenTracing integrations is very simple. For Spring Boot there is
+an auto-configuration which instruments all REST controllers and `RestTemplate` beans. Just add the following
+dependency to the classpath:
+
+----
+<dependency>
+    <groupId>io.opentracing.contrib</groupId>
+    <artifactId>opentracing-spring-web-autoconfigure</artifactId>
+</dependency>
+----
+
+This dependency requires only one thing and that is tracer bean which will be used to report data to the chosen
+tracing system. If we don't specify this bean auto-configuration will choose `NoopTracer`.
+
+Because we are using OpenTracing instrumentation we are not bound to any specific tracing system.
+We will now show how to first use Jaeger and then switch to Zipkin. We will see that changing tracing system is just
+a matter of configuration.
+
+As we mentioned the only tracing configuration needed here is to provide a tracer bean.
```

Comment:
The configuration is to provide a the tracer bean. It can be done via xml or java dsl